develop/parachains/testing/mock-runtime.md

```diff
@@ -0,0 +1,87 @@
+
+## Configuring a Mock Runtime
+
+### Testing Module
+
+The mock runtime includes all the necessary pallets and configurations needed for testing. To simplify the process, you can create a module that integrates all components, making it easier to assess how pallets and system elements interact.
+
+Here's a simple example of how to create a testing module that simulates these interactions:
+
+```rust
+--8<-- 'code/develop/parachains/testing/mock-runtime/integration-testing-module.rs'
+```
+
+!!! note
+    The `crate::*;` snippet imports all the components from your crate (including runtime configurations, pallet modules, and utility functions) into the `tests` module. This allows you to write tests without manually importing each piece, making the code more concise and readable.
+
+Alternatively, you can create a separate `mock.rs` file to define the configuration for your mock runtime and a companion `tests.rs` file to house the specific logic for each test.
+
+Once the testing module is configured, you can craft your mock runtime using the [`frame_support::runtime`](https://paritytech.github.io/polkadot-sdk/master/frame_support/attr.runtime.html){target=\_blank} macro. This macro allows you to define a runtime environment that will be created for testing purposes:
+
+```rust
+--8<-- 'code/develop/parachains/testing/mock-runtime/mock-runtime.rs'
+```
+### Genesis Storage
+
+The next step is configuring the genesis storage—the initial state of your blockchain. Genesis storage sets the starting conditions for the runtime, defining how pallets are configured before any blocks are produced.
+
+In Polkadot SDK, you can create this storage using the [`BuildStorage`](https://paritytech.github.io/polkadot-sdk/master/sp_runtime/trait.BuildStorage.html){target=\_blank} trait from the [`sp_runtime`](https://paritytech.github.io/polkadot-sdk/master/sp_runtime){target=\_blank} crate. This trait is essential for building the configuration that initializes the blockchain's state.
+
+The function `new_test_ext()` demonstrates setting up this environment. It uses `frame_system::GenesisConfig::<Test>::default()` to generate a default genesis configuration for the runtime, followed by `.build_storage()` to create the initial storage state. This storage is then converted into a format usable by the testing framework, [`sp_io::TestExternalities`](https://paritytech.github.io/polkadot-sdk/master/sp_io/type.TestExternalities.html){target=\_blank}, allowing tests to be executed in a simulated blockchain environment.
+
+Here's the code that sets the genesis storage configuration:
+
+```rust
+--8<-- 'code/develop/parachains/testing/mock-runtime/genesis-config.rs'
+```
+
+You can also customize the genesis storage to set initial values for your runtime pallets. For example, you can set the initial balance for accounts like this:
```

Comment:
what is critical and you are not mentioning is that you can only customize the fields that you define in your pallet via `#[pallet::genesis]`.

Ideally, this guide would take a few existing pallet, and one custom pallet (built in a previous tutorial), and stitch them together in a `mock`. This would check all the info boxes. 

Can this still be done?

Reply:
> what is critical and you are not mentioning is that you can only customize the fields that you define in your pallet via #[pallet::genesis].

@kianenigma, the `#[pallet::genesis]` macro is listed https://paritytech.github.io/polkadot-sdk/master/frame_support/pallet_macros/index.html.

Should I use `[#[pallet::genesis_build]](https://paritytech.github.io/polkadot-sdk/master/frame_support/pallet_macros/attr.genesis_build.html)` instead?

> Ideally, this guide would take a few existing pallet, and one custom pallet (built in a previous tutorial), and stitch them together in a mock. This would check all the info boxes.

I believe this topic will be covered in the tutorials section; this guide is part of the development section. This part isn’t intended to be a step-by-step article, we showcase key concepts here so that tutorials can leverage them

Reply:
fyi - genesis macros clarification added in [f7f1272](https://github.com/polkadot-developers/polkadot-docs/pull/248/commits/f7f12725a27645cdfd96e4e6ad2bba28a24c97f5)